### PR TITLE
Playbooks & scripts for building, registering and rotating octavia amphora images

### PR DESCRIPTION
**Pull request by jovial**
Taken from: https://github.com/stackhpc/kayobe-ops/pull/12

### CODE REVIEW

File: octavia-amphora-image-register.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    - name: Ensure Octavia Amphora image is registered
+      vars:
+        ansible_python_interpreter: "{{ venv }}/bin/python"
+      os_image:
+        auth_type: password
+        auth: "{{ openstack_auth }}"
+        ca_cert: "{{ openstack_cacert }}"
+        interface: "{{ openstack_interface }}"
+        name: amphora-x64-haproxy.qcow2
```
Comment (jovial) [inline octavia-amphora-image-register.yml]:
```suggestion
        name: amphora-x64-haproxy
```

File: octavia-amphora-image-register.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        properties:
+          hw_architecture: x86_64
+          hw_rng_model: virtio
+
+    # FIXME: Use 'tags' parameter of os_image module available from
+    # openstack.cloud.image 1.5.0.
+    - name: Ensure Octavia Amphora image is tagged
+      shell:
+        cmd: >-
+          {{ venv }}/bin/openstack image set amphora-x64-haproxy.qcow2 --tag amphora
```
Comment (jovial) [inline octavia-amphora-image-register.yml]:
```suggestion
          {{ venv }}/bin/openstack image set amphora-x64-haproxy --tag amphora
```

File: tools/loadbalancer-failover.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (jovial) [inline tools/loadbalancer-failover.sh:1]: Will move to scripts folder for consistency

Comment (jovial) [reply tools/loadbalancer-failover.sh:1]: Maybe tools folder is OK as this isn't technically related to the ansible

File: etc/kayobe/ansible/octavia-amphora-image-build.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            dest: "{{ src_path }}"
+            repo: "https://opendev.org/openstack/octavia"
+            version: "{{ openstack_branch }}"
+
+        - name: Install diskimage-builder in a virtual environment
+          pip:
+            name: diskimage-builder
+            # NOTE: Ussuri-Wallably upper constraints all fail to build the image.
+            # This is just for diskimage-builder, and doesn't affect the version of amphora.
+            extra_args: "-c https://opendev.org/openstack/requirements/raw/branch/stable/xena/upper-constraints.txt"
```
Comment (markgoddard) [inline etc/kayobe/ansible/octavia-amphora-image-build.yml]: What about yoga?

Comment (jovial) [reply etc/kayobe/ansible/octavia-amphora-image-build.yml]: Shall I change back to `{{ pip_upper_constraints_file }}`?

Comment (jovial) [reply etc/kayobe/ansible/octavia-amphora-image-build.yml]: Added `amphora_dib_upper_constraints_file` that defaults to `pip_upper_constraints_file`

**Comment by jovial** [comment]
Just going to fix that linting issue

File: etc/kayobe/ansible/octavia-amphora-image-build.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/ansible/octavia-amphora-image-build.yml:1]: I had hoped we've start using this collection for this kind of thing, but didn't push it enough: https://github.com/stackhpc/ansible-collection-openstack-ops

**Review by markgoddard — APPROVED** [review]

File: etc/kayobe/ansible/octavia-amphora-image-build.yml (hunk 5 above)
Comment (jovial) [reply etc/kayobe/ansible/octavia-amphora-image-build.yml:1]: Ahh - that sounds nice. Maybe we can move the code at somepoint.

**Pull request merged**